src/test/java/gui/TestEditRulePopup.java

```diff
@@ -0,0 +1,21 @@
+package test.java.gui;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+import main.java.gui.EditRulePopup;
+
+/**
+ * This test is validating the window size, to ensure the popup is being
+ * displayed properly.
+ */
+class TestEditRulePopup {

```

Comment:
estava a pensar.. (podes meter só TODO e passar à frente, não te mates mais neste PR) - tendo em conta as funcionalidades novas, não há mais nada que possa ser coberto por mais unit tests?

Reply:
Not sure... Por um lado, diria que o advanced mode tem bem mais possibilidades de implementar unit tests do que o resto do código, já que uma vez que é texto corrido os resultados são mais imprevisiveis... Ao mesmo tempo, não faço nenhum tipo de verificação quanto ao input aqui, essa verificação vai ser feita mais do lado do código em que a fran está a trabalhar, então... I'll have to look into it. Apesar disso, em termos de UI tive uns bugs estranhos em relação às JComboBoxes duplicarem as opções selecionaveis. Esses bugs entretanto já foi corrigidos, mas verificar se o número de elementos selecionaveis é o correto em cada uma das combo boxes pode ser uma opção razoável para aumentar a cobertura do código.

Reply:
sounds great :D achas que dá para fazer isso rápido? :)

Reply:
Eu voto actually em o Hugo só fazer o essencial pra terminar esse PR. Ele já está a trabalhar desde sempre nisso e estamos precisando mto dessa ui pros próximos tickets. O "só mais isso aqui, só mais ali" já juntou mais 1 semana de trabalho pra ele. Also, o meu ticket é o de fazer o store das regras, então os unit tests disso acho que fazem parte do meu ticket. Lembrem que a ui não guarda regra de negócio. O que vai acontecer é que o meu controlador não vai deixar salvar um regra não válida.

Reply:
sim mas tal como ele disse, a lista tem de apresentar o número certo de rules, e isso é do UI e deve ser testado. Mais vale fazer aqui ou por um TODO para fazer aqui, do que deixar por fazer e passar

Reply:
Se a parte lógica já está implementada e só falta os Tests Unit,  eu acredito que o melhor é fazer merge para podemos seguir com o trabalho. A parte do Hugo tem impacto e provoca dependencias com as próximas tarefas. Os testes fazemos quando as tarefas relacionadas estiverem implementadas. Eu concordo com a @ftfni-iscteiul que esta seja a melhor estratégia.